dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -982,6 +982,119 @@ TRASH_TrashFile(LPCWSTR wszPath)
     return DeleteFileToRecycleBin(wszPath);
 }
 
+static void TRASH_PlayEmptyRecycleBinSound()
+{
+    LPCWSTR lpRegSubKey = L"AppEvents\\Schemes\\Apps\\Explorer\\EmptyRecycleBin";
+    HKEY hRegKey = NULL;
+    HKEY hRegSnd = NULL;
+    LPWSTR lpRegVal = NULL;
+    LPWSTR lpSndPath = NULL;
+    DWORD dwRegValType, dwRegValSize, dwSndPathLen;
+    LONG lError;
+
+    /* Open registry key */
+    lError = RegOpenKeyExW(HKEY_CURRENT_USER, lpRegSubKey, 0, KEY_READ, &hRegKey);
```

Comment:
This is a cpp file, no need for all this manual closing /error handling, `CRegKey` for the registry and `CHeapPtr<WCHAR>` for the buffers should cut about 70% off from this function.